### PR TITLE
adding tekton task for hugo static site generator

### PR DESCRIPTION
**Pull request by shipit**
# Changes

Adds `hugo` task for static site generation. It is based on the [hugo project](https://github.com/gohugoio/hugo) and includes a sample `pipelineRun`. It is tested successfully with `./test/run-test.sh hugo 0.1`.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: task/hugo/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- **work-dir**: The directory where `hugo` will be run. It should contain the `config.toml` file and the `content` and `layouts` directories.
+
+## Results
+
+- `output`: The directory where the generated site is located.
+
+## Usage
+
+```
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/hugo/0.1/README.md:36]: can we start using `tekton.dev/v1`?

File: task/hugo/0.1/hugo.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/hugo/0.1/hugo.yaml:1]: same

File: task/hugo/0.1/hugo.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        - name: HUGO_MINIFY
+          value: $(params.minify)
+        - name: HUGO_BASEURL
+          value: $(params.base-url)
+        - name: HUGO_VERBOSE
+          value: $(params.verbose)
+        - name: HUGO_CONTENT_DIR
+          value: $(params.content-dir)
+
+      image: docker.io/hugomods/hugo:0.122.0
```
Comment (vinamra28) [inline task/hugo/0.1/hugo.yaml:62]: can we tag this with digest as well?

**Comment by vinamra28** [comment]
also @shipit, can you please squash the commits?

File: task/hugo/0.1/README.md (hunk 1 above)
Comment (shipit) [reply task/hugo/0.1/README.md]: yes, fixed

File: task/hugo/0.1/hugo.yaml (hunk 2 above)
Comment (shipit) [reply task/hugo/0.1/hugo.yaml]: fixed

File: task/hugo/0.1/hugo.yaml (hunk 3 above)
Comment (shipit) [reply task/hugo/0.1/hugo.yaml]: done

**Comment by shipit** [comment]
> also @shipit, can you please squash the commits?

done

**Review by dibyom — COMMENTED** [review]
/lgtm

**Pull request merged**